toolchain/check/decl_name_stack.h

```diff
@@ -165,6 +165,8 @@ class DeclNameStack {
   // Information about a declaration name that has been temporarily removed from
   // the stack and will later be restored. Names can only be suspended once they
   // are finished.
+  //
+  // This type is large, so moves of this type should be avoided.
```

Comment:
With this and `PendingThunk`, is using `MoveOnly` worthwhile? I think copies may be indirectly avoided due to the `ScopeStackEntry` use, but perhaps it's worth sprinkling around for explicitness?

Mainly I think of it in context of the comment because it'd make clear that moves are discouraged but explicitly supported (copies are right out).

Reply:
There's two subleties here: one is that these types are `MoveOnly` because they contain a `MoveOnly` member, and the other is that having multiple `MoveOnly` base classes within  this object (one indirectly in a member and another directly on the class) can lead to extra padding being added to the class in order to prevent them from having the same address.

We can fix the latter by making `MoveOnly` a CRTP base class (so the repeated `MoveOnly`s aren't the same type), but that makes the type less simple to use. I'm fine with that if you'd prefer we be more explicit here.

Reply:
Per discussion, leaning towards CRTP because it could occur by accident, and it's easy to miss as a concern